webflow-hacks/form-hacks/18/18.ts

```diff
@@ -0,0 +1,27 @@
+// when the DOM is ready
+document.addEventListener('DOMContentLoaded', function () {
+  // make an array of invalid domains
+  const invalidDomains = ['gmail.com', 'yahoo.com', 'hotmail.com', 'competitor.com'];
+  // set the fs-hacks selector
+  const SUBMIT_BUTTON_SELECTOR = '[fs-hacks-element="submit"]';
+  const EMAIL_SELECTOR = '[fs-hacks-element="email"]';
+  // get the dom elements
+  const submitBtn = document.querySelector<HTMLButtonElement>(SUBMIT_BUTTON_SELECTOR);
+  const emailInput = document.querySelector<HTMLInputElement>(EMAIL_SELECTOR);
+  // check if the elements exists
+  if (!submitBtn || !emailInput) return;
+  // on submit button click
+  submitBtn.addEventListener('click', function () {
+    // split email at '@' character to get domain
+    const domainPart = emailInput.value.split('@')[1];
+    // if the domain exists in the invalidDomains array
+    if (invalidDomains.indexOf(domainPart) !== -1) {
```

Comment:
There's a better way of checking this, using [Array.includes()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes):
`invalidDomains.includes(domainPart)`